### PR TITLE
Add support for QuickLogic EOS S3 SoC

### PR DESCRIPTION
**Pull request by PiotrZierhoffer**
This PR is the first from a series of PRs related to the EOS S3 ecosystem.

EOS S3 is a Cortex-M4-F based platform with an FPGA and a sensor processing subsystem.

The series will add support for the SoC itself, for the boards and several drivers.

More to read:

- QuickLogic EOS S3 description: https://www.quicklogic.com/products/eos-s3/
- QuickFeather board press release: https://antmicro.com/blog/2020/03/quickfeather-release/

The series will consist of the following PRs:

- [x] EOS S3 SoC
- [x] QuickLogic Chandalar board
- [ ] QuickLogic QuickFeather board
- [ ] GPIO support
- [ ] SPI support
- [ ] on-board FPGA support

### CODE REVIEW

File: west.yml [hunk 1]
```diff
@@ -21,6 +21,8 @@ manifest:
   remotes:
     - name: upstream
       url-base: https://github.com/zephyrproject-rtos
+    - name: antmicro
+      url-base: https://github.com/antmicro
```
Comment (tgorochowik) [inline west.yml]: This obviously cannot go in like this, please refer to: https://docs.zephyrproject.org/latest/guides/modules.html#submitting-a-new-module first.

Comment (carlescufi) [inline west.yml]: This is not allowed in the upstream manifest.  I can create the HAL repo for you instead

**Review by carlescufi — CHANGES_REQUESTED** [review]

File: west.yml (hunk 1 above)
Comment (PiotrZierhoffer) [reply west.yml]: That's what I assumed, but decided I'd create a PR first and change it later. A `hal_quicklogic` repo would be very useful indeed

**Comment by ioannisg** [comment]
Could you send out the Board definition in the same PR? So CI will pick up and build for the Board/SoC

**Review by ioannisg — COMMENTED** [review]
Thanks! Pls, add the Board support (Can be minimal) as part of this PR

**Comment by kowalewskijan** [comment]
I removed `QuickLogic Chandalar board` and replaced it with `QuickLogic QuickFeather board` since it has much more features enabled, which will come in next PRs.

**Comment by PiotrZierhoffer** [comment]
As we changed the board we use, I ask for re-review of this PR

**Review by tgorochowik — APPROVED** [review]
Looks good, should be ready to merge once https://github.com/zephyrproject-rtos/hal_quicklogic/pull/1 is in.

**Comment by kowalewskijan** [comment]
Since the [HAL PR](zephyrproject-rtos/hal_quicklogic#1) was merged, I updated `west.yml` manifest to point to correct remote and commit.

**Comment by PiotrZierhoffer** [comment]
@carlescufi @ioannisg as from our perspective this is good to go, would you care to [re]review this?

File: soc/arm/quicklogic_eos_s3/soc.c [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	CRU->C11_CLK_GATE = C11_CLK_GATE_PATH_0_ON;
+
+	/* Set divider for domain C11 to ~ 5.12MHz */
+	CRU->CLK_CTRL_D_0 = (CLK_CTRL_CLK_DIVIDER_ENABLE |
+		CLK_CTRL_CLK_DIVIDER_RATIO_12);
+}
+
+
+
+static int eos_s3_init(struct device *arg)
```
Comment (carlescufi) [inline soc/arm/quicklogic_eos_s3/soc.c:67]:
```suggestion
static int eos_s3_init(const struct device *arg)
```

**Comment by carlescufi** [comment]
This is almost ready for merging, just one check, will this be built by default CI?

**Comment by carlescufi** [comment]
@galak and @MaureenHelm please take a look and consider it for 2.4.

**Review by carlescufi — COMMENTED** [review]
I have no objections except for:

- Fix the `const struct device`
- Make sure this board is built in CI

File: soc/arm/quicklogic_eos_s3/soc.c (hunk 2 above)
Comment (kowalewskijan) [reply soc/arm/quicklogic_eos_s3/soc.c]: Fixed the line as suggested

**Comment by kowalewskijan** [comment]
I fixed the suggestion from @carlescufi and rebased the branch.
@carlescufi : I can see that in the `buildkite/zephyr` CI there is a `quick_feather` target called in tests e.g.: [here](https://buildkite.com/zephyr/zephyr/builds/7457#5e92a630-dc93-41a6-a1dd-b88d4fca23e9/516-529), so I assume it is fine.

**Review by carlescufi — APPROVED** [review]

**Review by nashif — APPROVED** [review]

**Comment by Willmish** [comment]
Hi @PiotrZierhoffer , me and @Jduchniewicz are working on a project utilising an EOS S3 based board (SparkFun's Qt PLUS), and are struggling to get it to flash properly using the adapted version of TinyFPGA programmer (https://github.com/QuickLogic-Corp/TinyFPGA-Programmer-Application/tree/master). I was wondering whether you could share with us what was your workflow/tools when working with the board?

 (Right now for us the board hangs when Verifying binary written to the board - this is happening on 2 different boards and tested on 3 different setups (MacOS, Ubuntu 20.04, Arch Linux)).
Screenshot 2023-05-27 at 16 37 53

Any help would be highly appreciated!

**Comment by Willmish** [comment]
For context - I also tried flashing using openOCD via STLINK but am always met with a time out when trying to halt the target CPU: 
Screenshot 2023-05-27 at 20 48 52
(Had to modify the board/quicklogic_quickfeather.cfg to use "hla_swd" instead of "swd" for the transport)

Also I belive @JDuchniewicz tried flashing using JLINK, without much success.

**Pull request merged**